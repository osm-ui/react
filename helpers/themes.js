import _ from 'lodash';
import { lighten } from 'polished';


export function buildDarkThemeConfig(baseConfig, colors, colorVariant) {
    const color1 = colors[`${colorVariant}1`];
    const color2 = colors[`${colorVariant}2`];
    const color3 = colors[`${colorVariant}3`];
    // const color4 = colors[`${colorVariant}4`];
    const color5 = colors[`${colorVariant}5`];

    return _.merge({}, baseConfig, {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color1,
        loaderColor: colors.white,
        controlColor: color1,
        hoverControlColor: colors.white,

        column: {
            nav: {
                color: colors.white,
                backgroundColor: lighten(0.1, color2),
                hoverColor: colors.white,
                hoverBackgroundColor: color3,
            },
        },
        toolbar: {
            button: {
                color: colors.white,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        titlebar: {
            color: colors.white,
            backgroundColor: color2,
            borderColor: color2,

            button: {
                color: colors.white,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        form: {
            button: {
                color: colors.white,
                backgroundColor: color2,
                borderColor: color5,
                hoverBackgroundColor: color3,
                hoverBorderColor: color5,
                focusBackgroundColor: color3,
                focusBorderColor: color5,
                activeBackgroundColor: color3,
                activeBorderColor: color5,

                primary: {
                    color: color2,
                    backgroundColor: colors.white,
                    borderColor: colors.white,
                    hoverBackgroundColor: colors.lightGray5,
                    hoverBorderColor: colors.lightGray5,
                    focusBackgroundColor: colors.lightGray5,
                    focusBorderColor: colors.lightGray5,
                    activeBackgroundColor: colors.lightGray5,
                    activeBorderColor: colors.lightGray5,
                },
            },
        },
    });
}


export function buildLightThemeConfig(baseConfig, colors, colorVariant) {
    const color1 = colors[`${colorVariant}1`];
    const color2 = colors[`${colorVariant}2`];
    const color3 = colors[`${colorVariant}3`];
    const color4 = colors[`${colorVariant}4`];
    const color5 = colors[`${colorVariant}5`];

    return _.merge({}, baseConfig, {
        color: colors.anthracite1,
        backgroundColor: color2,
        borderColor: color1,
        loaderColor: colors.anthracite1,
        controlColor: color4,
        hoverControlColor: colors.anthracite1,

        column: {
            nav: {
                color: colors.anthracite1,
                backgroundColor: lighten(0.2, color2),
                hoverColor: colors.anthracite1,
                hoverBackgroundColor: color3,
            },
        },
        toolbar: {
            button: {
                color: colors.anthracite1,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        titlebar: {
            color: colors.anthracite1,
            backgroundColor: color2,
            borderColor: color2,

            button: {
                color: colors.anthracite1,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        form: {
            button: {
                color: colors.anthracite1,
                backgroundColor: color2,
                borderColor: color5,
                hoverBackgroundColor: color3,
                hoverBorderColor: color5,
                focusBackgroundColor: color3,
                focusBorderColor: color5,
                activeBackgroundColor: color3,
                activeBorderColor: color5,

                primary: {
                    color: color2,
                    backgroundColor: colors.anthracite1,
                    borderColor: colors.anthracite1,
                    hoverBackgroundColor: colors.lightGray5,
                    hoverBorderColor: colors.lightGray5,
                    focusBackgroundColor: colors.lightGray5,
                    focusBorderColor: colors.lightGray5,
                    activeBackgroundColor: colors.lightGray5,
                    activeBorderColor: colors.lightGray5,
                },
            },
        },
    });
}


export default {
    buildDarkThemeConfig,
    buildLightThemeConfig,
};
