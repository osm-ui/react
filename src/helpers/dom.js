export function isOverflowYHidden(element) {
  return window.getComputedStyle(element).overflowY !== 'hidden';
}
