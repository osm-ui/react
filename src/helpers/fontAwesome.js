export function makePrefix(style) {
  return `fa${style[0]}`;
}

export function makeName(name) {
  return `fa-${name}`;
}
