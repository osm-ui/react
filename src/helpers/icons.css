@import url('https://use.fontawesome.com/releases/v5.2.0/css/all.css');
