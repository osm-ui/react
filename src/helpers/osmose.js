export function formatDeletedTags(fix, osmTags) {
  return {
    ...fix,
    del: fix.del.map(deletedTag => {
      return osmTags.find(tag => tag.k === deletedTag);
    })
  };
}

export function formatOsmTags(data) {
  return data.reduce((acc, item) => {
    return {
      ...acc,
      [item.k]: item.v
    };
  }, {});
}

export function fixOsmTags(osmTags, fix) {
  const newOsm = {
    ...formatOsmTags(osmTags),
    ...formatOsmTags(fix.add),
    ...formatOsmTags(fix.mod)
  };

  fix.del.forEach(tag => Reflect.deleteProperty(newOsm, tag));

  return newOsm;
}
