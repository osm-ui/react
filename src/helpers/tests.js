import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import DefaultTheme from 'themes/Default';

export const snapshotWithElementChildren = (TestedElement, props) => {
  const result = shallow(
    <DefaultTheme>
      <TestedElement {...props}>
        <div />
      </TestedElement>
    </DefaultTheme>
  );

  expect(result.childAt(0)).toMatchSnapshot();
};

export const snapshotWithTextChildren = (TestedElement, props) => {
  const result = shallow(
    <DefaultTheme>
      <TestedElement {...props}>This is a text</TestedElement>
    </DefaultTheme>
  );

  expect(result.childAt(0)).toMatchSnapshot();
};
