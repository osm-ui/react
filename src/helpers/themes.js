import _ from 'lodash';
import { lighten, darken } from 'polished';
import { injectGlobal } from 'styled-components';

export function buildDarkThemeConfig(baseConfig, colors, colorVariant) {
  const color1 = colors[`${colorVariant}1`];
  const color2 = colors[`${colorVariant}2`];
  const color3 = colors[`${colorVariant}3`];
  // const color4 = colors[`${colorVariant}4`];
  const color5 = colors[`${colorVariant}5`];

  /* eslint-disable no-unused-expressions */
  injectGlobal`
    .osm-ui-react-marker-${colorVariant} {
      color: ${colors.white};

      #colorized, #colorized * {
        fill: ${color2} !important;
      }
    }
  `;
  /* eslint-enable */

  return _.merge({}, baseConfig, {
    color: colors.white,
    backgroundColor: color2,
    borderColor: color1,
    loaderColor: colors.white,
    controlColor: color1,
    hoverControlColor: colors.white,

    sidebar: {
      nav: {
        color: colors.white,
        backgroundColor: lighten(0.1, color2),
        hoverColor: colors.white,
        hoverBackgroundColor: color3
      }
    },
    toolbar: {
      button: {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    titlebar: {
      color: colors.white,
      backgroundColor: color2,
      borderColor: color2,

      button: {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    form: {
      button: {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color5,
        hoverBackgroundColor: color3,
        hoverBorderColor: color5,
        focusBackgroundColor: color3,
        focusBorderColor: color5,
        activeBackgroundColor: color3,
        activeBorderColor: color5,

        primary: {
          color: color2,
          backgroundColor: colors.white,
          borderColor: colors.white,
          hoverBackgroundColor: colors.lightGray5,
          hoverBorderColor: colors.lightGray5,
          focusBackgroundColor: colors.lightGray5,
          focusBorderColor: colors.lightGray5,
          activeBackgroundColor: colors.lightGray5,
          activeBorderColor: colors.lightGray5
        }
      }
    }
  });
}

export function buildLightThemeConfig(baseConfig, colors, colorVariant) {
  const color1 = colors[`${colorVariant}1`];
  const color2 = colors[`${colorVariant}2`];
  const color3 = colors[`${colorVariant}3`];
  const color4 = colors[`${colorVariant}4`];
  const color5 = colors[`${colorVariant}5`];
  const darkColor1 = darken(0.27, color1);

  /* eslint-disable no-unused-expressions */
  injectGlobal`
        .osm-ui-react-marker-${colorVariant} {
            color: ${darkColor1};

            #colorized, #colorized * {
                fill: ${color2} !important;
            }
        }
    `;
  /* eslint-enable */

  return _.merge({}, baseConfig, {
    color: darkColor1,
    backgroundColor: color2,
    borderColor: color1,
    loaderColor: darkColor1,
    controlColor: color4,
    hoverControlColor: darkColor1,

    sidebar: {
      nav: {
        color: darkColor1,
        backgroundColor: color3,
        hoverColor: darkColor1,
        hoverBackgroundColor: lighten(0.14, color3)
      }
    },
    toolbar: {
      button: {
        color: darkColor1,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    titlebar: {
      color: darkColor1,
      backgroundColor: color2,
      borderColor: color2,

      button: {
        color: darkColor1,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    form: {
      button: {
        color: darkColor1,
        backgroundColor: color2,
        borderColor: color5,
        hoverBackgroundColor: color3,
        hoverBorderColor: color5,
        focusBackgroundColor: color3,
        focusBorderColor: color5,
        activeBackgroundColor: color3,
        activeBorderColor: color5,

        primary: {
          color: color2,
          backgroundColor: darkColor1,
          borderColor: darkColor1,
          hoverBackgroundColor: colors.lightGray5,
          hoverBorderColor: colors.lightGray5,
          focusBackgroundColor: colors.lightGray5,
          focusBorderColor: colors.lightGray5,
          activeBackgroundColor: colors.lightGray5,
          activeBorderColor: colors.lightGray5
        }
      }
    }
  });
}

export default {
  buildDarkThemeConfig,
  buildLightThemeConfig
};
