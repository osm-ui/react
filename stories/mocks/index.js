export const osmData = {
  original: {
    name: "Romain",
    age: '32',
    gender: "Male",
    eyes: "Green",
    profession: "Doctor"
  },
  fixed: {
    name: "Romain",
    address: "11000 VeryFarAwayStreet, 98765 VeryCoolCity",
    age: '32',
    gender: "Male",
    eyes: "Blue",
    hasCat: 'yes',
  }
};
