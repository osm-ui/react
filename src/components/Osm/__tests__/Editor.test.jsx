import React from 'react';
import 'jest-styled-components';
import { snapshotWithElementChildren } from 'helpers/tests';
import { Editor } from 'index';
import { osmData } from '../../../../stories/mocks/index';

describe('When using snapshots', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(Editor, {
      original: osmData.original,
      fixed: osmData.fixed
    }));
});
