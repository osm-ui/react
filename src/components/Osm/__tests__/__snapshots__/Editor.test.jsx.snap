// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using snapshots Should render with an element children 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;
}

.c4 {
  border-style: solid;
  border-width: 2px;
  border-radius: 4px;
}

.c4.btn-default {
  font-weight: 500;
  color: #3b3f45;
  background-color: #ffffff;
  border-color: #c9cacc;
}

.c4.btn-default:hover {
  color: #3b3f45;
  background-color: #eceeef;
  border-color: #c9cacc;
}

.c4.btn-default:focus {
  color: #3b3f45;
  background-color: #eceeef;
  border-color: #c9cacc;
}

.c4.btn-default:active,
.c4.btn-default .active {
  color: #3b3f45;
  background-color: #eceeef;
  border-color: #c9cacc;
}

.c4.btn-primary {
  font-weight: 500;
  color: #ffffff;
  background-color: #2196f3;
  border-color: #186fc5;
}

.c4.btn-primary:hover {
  color: #ffffff;
  background-color: #71bcf7;
  border-color: #2196f3;
}

.c4.btn-primary:focus {
  color: #ffffff;
  background-color: #71bcf7;
  border-color: #2196f3;
}

.c4.btn-primary:active,
.c4.btn-primary .active {
  color: #ffffff;
  background-color: #71bcf7;
  border-color: #2196f3;
}

.c4.btn-info {
  font-weight: 500;
  color: #000a0a;
  background-color: #00c2b8;
  border-color: #00948d;
}

.c4.btn-info:hover {
  color: #000a0a;
  background-color: #5cd7d1;
  border-color: #00c2b8;
}

.c4.btn-info:focus {
  color: #000a0a;
  background-color: #5cd7d1;
  border-color: #00c2b8;
}

.c4.btn-info:active,
.c4.btn-info .active {
  color: #000a0a;
  background-color: #5cd7d1;
  border-color: #00c2b8;
}

.c4.btn-success {
  font-weight: 500;
  color: #ffffff;
  background-color: #4caf50;
  border-color: #348538;
}

.c4.btn-success:hover {
  color: #ffffff;
  background-color: #7cc47f;
  border-color: #4caf50;
}

.c4.btn-success:focus {
  color: #ffffff;
  background-color: #7cc47f;
  border-color: #4caf50;
}

.c4.btn-success:active,
.c4.btn-success .active {
  color: #ffffff;
  background-color: #7cc47f;
  border-color: #4caf50;
}

.c4.btn-warning {
  font-weight: 500;
  color: #543500;
  background-color: #ffa700;
  border-color: #de8c00;
}

.c4.btn-warning:hover {
  color: #543500;
  background-color: #ffd659;
  border-color: #ffa700;
}

.c4.btn-warning:focus {
  color: #543500;
  background-color: #ffd659;
  border-color: #ffa700;
}

.c4.btn-warning:active,
.c4.btn-warning .active {
  color: #543500;
  background-color: #ffd659;
  border-color: #ffa700;
}

.c4.btn-danger {
  font-weight: 500;
  color: #ffffff;
  background-color: #f44336;
  border-color: #c92d2d;
}

.c4.btn-danger:hover {
  color: #ffffff;
  background-color: #f8877f;
  border-color: #f44336;
}

.c4.btn-danger:focus {
  color: #ffffff;
  background-color: #f8877f;
  border-color: #f44336;
}

.c4.btn-danger:active,
.c4.btn-danger .active {
  color: #ffffff;
  background-color: #f8877f;
  border-color: #f44336;
}

.c4.btn-link {
  font-weight: 400;
  color: #2196f3;
  background-color: transparent;
  border-color: transparent;
}

.c4.btn-link:hover {
  color: #2196f3;
  background-color: transparent;
  border-color: transparent;
}

.c4.btn-link:focus {
  color: #2196f3;
  background-color: transparent;
  border-color: transparent;
}

.c4.btn-link:active,
.c4.btn-link .active {
  color: #2196f3;
  background-color: transparent;
  border-color: transparent;
}

.c4.btn-xs {
  font-size: 1.2rem;
  padding-left: 1.4rem;
  padding-right: 1.4rem;
}

.c4.btn-sm {
  padding-left: 1.6rem;
  padding-right: 1.6rem;
}

.c4.btn-md {
  padding-left: 2rem;
  padding-right: 2rem;
}

.c4.btn-lg {
  padding-left: 2.4rem;
  padding-right: 2.4rem;
}

.c4.shape-round.btn-xs {
  border-radius: 1.25rem;
}

.c4.shape-round.btn-sm {
  border-radius: 1.7rem;
}

.c4.shape-round.btn-md {
  border-radius: 2.1rem;
}

.c4.shape-round.btn-lg {
  border-radius: 2.45rem;
}

.c8 {
  opacity: 1;
  background-color: #ffffff;
  border-color: #c9cacc;
  border-width: 2px;
  border-radius: 0;
  box-shadow: inset 0 1px 1px rgba(0,0,0,0.075);
}

.c8.form-control {
  height: 100%;
}

.c8:focus,
.c8.form-control:focus {
  opacity: 1;
  background-color: #ffffff;
  border-color: #2196f3;
  box-shadow: inset 0 1px 1px rgba(0,0,0,0.075);
}

.c8:disabled {
  opacity: 0.6;
}

.c6 {
  opacity: 1;
  resize: vertical;
  background-color: #ffffff;
  border-color: #c9cacc;
  border-width: 2px;
  border-radius: 0;
  box-shadow: inset 0 1px 1px rgba(0,0,0,0.075);
}

.c6.form-control {
  height: 100%;
}

.c6:focus,
.c6.form-control:focus {
  opacity: 1;
  background-color: #ffffff;
  border-color: #2196f3;
  box-shadow: inset 0 1px 1px rgba(0,0,0,0.075);
}

.c6:disabled {
  opacity: 0.6;
}

.c2 {
  margin-bottom: 2rem;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

.c2 .main {
  z-index: 1;
  position: relative;
  background-color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .main .c3 {
  border-radius: 4px 0 0 4px;
}

.c2 .main .value {
  -webkit-flex: 1 0 10rem;
  -ms-flex: 1 0 10rem;
  flex: 1 0 10rem;
  line-height: 3rem;
  padding: 1px 0;
  padding-left: 1rem;
  border: 1px solid #c9cacc;
  border-left: none;
  border-radius: 0 4px 4px 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c2 .main .actions {
  position: absolute;
  opacity: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  right: 0;
  top: 0;
  bottom: 0;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

.c2 .main .actions .action {
  color: #ffffff;
  line-height: 3rem;
  border: 2px solid transparent;
  cursor: pointer;
}

.c2 .main .actions .del {
  background-color: #f44336;
  border-color: #c92d2d;
  border-radius: 0 4px 0 0;
}

.c2 .main .actions .mod {
  background-color: #ffa700;
  border-color: #de8c00;
  border-radius: 0;
}

.c2.add .c3,
.c2.add .value {
  border-color: #7cc47f;
}

.c2.add .c3 {
  color: #ffffff;
  border-color: #348538;
  background-color: #7cc47f;
}

.c2.mod .c3,
.c2.mod .value {
  border-color: #ffd659;
}

.c2.mod .c3 {
  color: #ffffff;
  border-color: #de8c00;
  background-color: #ffa700;
}

.c2.del .c3,
.c2.del .value {
  border-color: #f8877f;
}

.c2.del .c3 {
  color: #ffffff;
  border-color: #c92d2d;
  background-color: #f8877f;
}

.c2.missing .value {
  color: #c92d2d;
  font-style: italic;
}

.c2.inactive,
.c2.removed {
  opacity: 0.2;
}

.c2.inactive .input-container,
.c2.removed .input-container,
.c2.inactive .textarea-container,
.c2.removed .textarea-container {
  visibility: hidden;
}

.c2 .input-container,
.c2 .textarea-container {
  height: 3.4rem;
  z-index: 0;
  margin-top: -3.4rem;
  -webkit-transition: margin-top 300ms ease-in-out,height 300ms ease-in-out;
  transition: margin-top 300ms ease-in-out,height 300ms ease-in-out;
}

.c2 .c7,
.c2 .c5 {
  line-height: 1.7rem;
  border-radius: 0 0 4px 4px;
}

.c2.selected {
  opacity: 1;
}

.c2.selected .c3 {
  border-radius: 4px 0 0 0;
}

.c2.selected .input-container,
.c2.selected .textarea-container {
  margin-top: 0;
}

.c2.selected .textarea-container {
  height: 8rem;
}

.c2.selected .value {
  border-radius: 0 4px 0 0;
}

.c2 .c3 {
  padding: 0 1rem;
}

.c2 .c3:focus {
  outline: none;
}

.c1 {
  max-width: 40rem;
}

.c1 .removedList {
  margin-top: 3rem;
}

.c1 .removedList > div:first-child {
  margin-bottom: 1rem;
  font-size: 2rem;
  font-weight: bold;
}

.c1 .add-item {
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

.c1 .add-item .input-container {
  display: inline-block;
}

.c1 .add-item.inactive {
  opacity: 0.2;
}

.c1 .add-item .add-button-container {
  display: inline-block;
  vertical-align: top;
  background-color: #ffffff;
  margin-left: -15rem;
  -webkit-transition: margin-left 300ms ease-in-out,width 300ms ease-in-out;
  transition: margin-left 300ms ease-in-out,width 300ms ease-in-out;
}

.c1 .add-item .input-container,
.c1 .add-item .add-button-container {
  width: 15rem;
}

.c1 .add-item span {
  -webkit-transition: -webkit-transform 300ms ease-in-out;
  -webkit-transition: transform 300ms ease-in-out;
  transition: transform 300ms ease-in-out;
}

.c1 .add-item.open .c3 {
  border-radius: 0 4px 4px 0;
}

.c1 .add-item.open span {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c1 .add-item.open .add-button-container {
  width: 0;
  margin-left: 0;
}

.c1 .add-item .c7 {
  opacity: 1;
  height: 3.4rem;
  font-size: 1.4rem;
  font-weight: 500;
  color: #ffffff;
  background-color: #71bcf7;
  border-color: #186fc5;
  border-right: none;
  border-radius: 4px 0 0 4px;
}

.c1 .add-item .c7::-webkit-input-placeholder {
  color: #c2e2fb;
  font-weight: normal;
}

.c1 .add-item .c7::-moz-placeholder {
  color: #c2e2fb;
  font-weight: normal;
}

.c1 .add-item .c7:-ms-input-placeholder {
  color: #c2e2fb;
  font-weight: normal;
}

.c1 .add-item .c7::placeholder {
  color: #c2e2fb;
  font-weight: normal;
}

.c1 .c3 {
  padding: 0 1rem;
}

.c1 .c3:focus {
  outline: none;
}

.c1 .c3.submit {
  margin-top: 5rem;
}

<span
  className="c0"
>
  <div
    className="c1"
  >
    <form>
      <div
        className="add c2"
      >
        <div
          className="main"
        >
          <button
            className="tag btn shape-square btn-default btn-sm c3 c4"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            address
          </button>
          <div
            className="value"
          >
            11000 VeryFarAwayStreet, 98765 VeryCoolCity
          </div>
        </div>
        <div
          className="textarea-container"
          style={null}
        >
          <textarea
            className="form-control c5 c6"
            disabled={false}
            onChange={[Function]}
            placeholder="Veuillez entrer une valeur"
            rows={6}
            value="11000 VeryFarAwayStreet, 98765 VeryCoolCity"
          />
          
        </div>
      </div>
      <div
        className="c2"
      >
        <div
          className="main"
        >
          <button
            className="tag btn shape-square btn-default btn-sm c3 c4"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            age
          </button>
          <div
            className="value"
          >
            32
          </div>
        </div>
        <div
          className="input-container"
          style={null}
        >
          <input
            className="form-control c7 c8"
            disabled={false}
            onChange={[Function]}
            placeholder="Veuillez entrer une valeur"
            type="text"
            value="32"
          />
          
        </div>
      </div>
      <div
        className="mod c2"
      >
        <div
          className="main"
        >
          <button
            className="tag btn shape-square btn-default btn-sm c3 c4"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            eyes
          </button>
          <div
            className="value"
          >
            Blue
          </div>
        </div>
        <div
          className="input-container"
          style={null}
        >
          <input
            className="form-control c7 c8"
            disabled={false}
            onChange={[Function]}
            placeholder="Veuillez entrer une valeur"
            type="text"
            value="Blue"
          />
          
        </div>
      </div>
      <div
        className="c2"
      >
        <div
          className="main"
        >
          <button
            className="tag btn shape-square btn-default btn-sm c3 c4"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            gender
          </button>
          <div
            className="value"
          >
            Male
          </div>
        </div>
        <div
          className="input-container"
          style={null}
        >
          <input
            className="form-control c7 c8"
            disabled={false}
            onChange={[Function]}
            placeholder="Veuillez entrer une valeur"
            type="text"
            value="Male"
          />
          
        </div>
      </div>
      <div
        className="add c2"
      >
        <div
          className="main"
        >
          <button
            className="tag btn shape-square btn-default btn-sm c3 c4"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            hasCat
          </button>
          <div
            className="value"
          >
            yes
          </div>
        </div>
        <div
          className="input-container"
          style={null}
        >
          <input
            className="form-control c7 c8"
            disabled={false}
            onChange={[Function]}
            placeholder="Veuillez entrer une valeur"
            type="text"
            value="yes"
          />
          
        </div>
      </div>
      <div
        className="c2"
      >
        <div
          className="main"
        >
          <button
            className="tag btn shape-square btn-default btn-sm c3 c4"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            name
          </button>
          <div
            className="value"
          >
            Romain
          </div>
        </div>
        <div
          className="input-container"
          style={null}
        >
          <input
            className="form-control c7 c8"
            disabled={false}
            onChange={[Function]}
            placeholder="Veuillez entrer une valeur"
            type="text"
            value="Romain"
          />
          
        </div>
      </div>
    </form>
    <div
      className="add-item"
    >
      <div
        className="input-container"
        style={null}
      >
        <input
          className="form-control c7 c8"
          disabled={false}
          onChange={[Function]}
          onClick={[Function]}
          placeholder="Entrez un tag"
          type="text"
          value=""
        />
        
      </div>
      <div
        className="add-button-container"
      >
        <button
          className="btn shape-square btn-primary btn-sm c3 c4"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden={true}
            className="fa fa-plus"
          />
        </button>
      </div>
    </div>
    <div
      className="removedList"
    >
      <div>
        Removed items
      </div>
      <div
        className="del inactive removed c2"
      >
        <div
          className="main"
        >
          <button
            className="tag btn shape-square btn-default btn-sm c3 c4"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            profession
          </button>
          <div
            className="value"
          >
            Doctor
          </div>
        </div>
      </div>
    </div>
    <button
      className="submit btn shape-round btn-default btn-md btn-block c3 c4"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Valider
    </button>
  </div>
</span>
`;
